SecretManagement.LastPass.psm1

```diff
@@ -216,6 +216,85 @@ function Get-VaultParams {
 
 }
 
+<#
+.SYNOPSIS
+Show LastPass Grid view secrets then show the selected secret.
+
+.DESCRIPTION
+Show LastPass Grid view secrets then show the selected secret. 
+
+.PARAMETER Vault
+Name of the vault used for the lookup
+
+.PARAMETER Filter
+Pre-filter secrets based on the specified keywords
+
+.PARAMETER KeepOpen
+If set, the secrets GridView will be automatically reloaded after a secret is shown.
+
+.PARAMETER PassThru
+If set, Secret will be returned as is without formatting.
+
+.EXAMPLE
+Show-LastPassConsoleGridView -Vault MyVault -KeepOpen 
+
+.NOTES
+This cmdlet can make use of the improved Out-ConsoleGridView cmdlet if using Powershell 6.2 or newer and  Microsoft.PowerShell.ConsoleGuiTools is installed.
+Otherwise, Out-GridView will be used.
+#>
+Function Show-LastPassConsoleGridView {
+    [CmdletBinding(DefaultParameterSetName='Default')]
+    Param(
+        [String]$Vault,
+        [String]$Filter = '*',
+        [Parameter(ParameterSetName = 'KeepOpen')]
+        [Switch]$KeepOpen,
+        [Parameter(ParameterSetName = 'PassThru')]
+        [Switch]$PassThru
```

Comment:
Why do you need this `PassThru`? I think it should be the default behavior to emit the result of the choice to the pipeline.

Reply:
Yeah, I can do that. 
My beef was / is with the multiline notes value. 
https://github.com/TylerLeonhardt/SecretManagement.LastPass/issues/24

I can't figure out (yet) how to apply a view formatter for a Idictionary 
(This is the only topic where I find there's a lack of examples to do advanced stuff with)

Ideally, I'd have a view formatter in place to auto-expand that property.
I can remove PassThru and instead replace it with a parameter to do the opposite.

Reply:
@TylerLeonhardt What do you think if I do the exact opposite of that commit. 
Meaning `-Passthru` become `-Formatted`. 
By default (without switch), secret is returned as is, perfect for variable assignement.

`Formatted` will include the extra treatment, which essentially make sure that the `Notes` field is expanded and readable from the output. 


```
 # By default, return the secret as is, everything will be fine
            if (!$Formatted) {return $Secret}

            # Intended for view, not for assignement. Add secret title and expand multiline notes in the console.
                Write-host $_.Name -ForegroundColor Cyan
                # Expand multiline notes (Ideally replaced by a format view at some point)
                if ($null -ne $Secret.Notes -and $Secret.Notes.IndexOf("`n") -ne -1) {
                    $Notes = $Secret.Notes
                    $Secret.Remove('Notes')
                    $Secret
                    "Notes:`n$Notes"
                }
                else {
                    $Secret
                }
```

Essentially, this provide the kind of readibility that using a view formatter on a collection would do (#24) except that I am not sure how to do a view formatter on a collection. 

I would be able to manage on a PSObject but since we have case-sensitivity, I need to stick to ordered dictionaries. 

If I could have a type or view formatter that expan the Notes field, then all that Formatted bit would be superfluous.

Reply:
tl;dr : I'd remove the PassThru, create a Formatted switch so that it reverse the default output.
Default output would be raw secret. 
Formatted output would ensure Notes field is expanded so you can read it from the screen.
All of that because I can't come with a way (yet) to solve #24

Reply:
> I would be able to manage on a PSObject but since we have case-sensitivity, I need to stick to ordered dictionaries.

You can cast an OrderedDictionary to a PSObject. `[PSObject]$dict` or `[pscustomobject]$dict`. Technically all objects in PowerShell are secretly a PSObject.

Reply:
Hmmm ... Good point. 
I'll check if I can do what I want from that. 

If I can have a LastPassSecret type inheriting from ordered dictionary and set a view to it so it return the Notes expanded by default without the need of "Format output altering" command, then it would be golden.

Reply:
Something they did in [PowerShell for GitHub](https://github.com/Microsoft/PowerShellForGitHub) is [have their Format.ps1xmls as normal](https://github.com/microsoft/PowerShellForGitHub/blob/master/Formatters/GitHubGists.Format.ps1xml) and then tack on the type name like so:

```
$item.PSObject.TypeNames.Insert(0, $TypeName)
```
https://github.com/microsoft/PowerShellForGitHub/blob/9600fc21120e17241e60606c5de3459d973026bb/GitHubGists.ps1#L1750

Reply:
Ah Thanks. 
I was playing around this concept just now. 

Your "Everything is a PSObject" quote put me on track.
I knew that but it was buried somewhere. 

```
$test  = [System.Collections.Specialized.OrderedDictionary]::new([System.StringComparer]::CurrentCulture)
$test.Add('ss','aa')
$test.Add('aaa','aa')
$test.Add('aaA', 'aa')
$test.PSObject.TypeNames.Insert(0,'LastPassSecret')
```

I am trying to flesh out some kind of ps1xml that would work and playing around with EzOut helper.


```
Write-FormatView -TypeName 'LastPassSecret' -Property 'Name','Value' -Width 1,1 | Out-FormatData | Add-FormatData; $test
```

```
<Configuration>
  <ViewDefinitions>
    <View>
      <Name>LastPassSecret</Name>
      <ViewSelectedBy>
        <TypeName>LastPassSecret</TypeName>
      </ViewSelectedBy>
      <TableControl>
        <TableHeaders>
          <TableColumnHeader>
            <Alignment>left</Alignment>
            <Width>1</Width>
          </TableColumnHeader>
          <TableColumnHeader>
            <Alignment>left</Alignment>
            <Width>1</Width>
          </TableColumnHeader>
        </TableHeaders>
        <TableRowEntries>
          <TableRowEntry>
            <TableColumnItems>
              <TableColumnItem>
                <PropertyName>Name</PropertyName>
              </TableColumnItem>
              <TableColumnItem>
                <PropertyName>Value</PropertyName>
              </TableColumnItem>
            </TableColumnItems>
          </TableRowEntry>
        </TableRowEntries>
      </TableControl>
    </View>
  </ViewDefinitions>
</Configuration>
```

It's not working yet though.
My thought is that if I can mess up the width of the column, I'll have succeeded in getting the proper tree syntax and I'll then dig around the Scriptblock property to get this working.

Maybe I'm owed a visit to the format.xsd schema to see if I can gain more insight in the proper manner to write something that would affect the collection.

Reply:
I was unsuccessful with the view formatter. 
This is a concept I'm slowly wrapping my head around but I find there's a lack of content compared to anything else on the matter so that's something I'll look into implementing at a latter time, if I can figure out the proper way to write the formatter. 

https://github.com/TylerLeonhardt/SecretManagement.LastPass/pull/23/commits/f695769b3ea13a3a53cadef962c1645b1bcf1158